sycl/plugins/cuda/CMakeLists.txt

```diff
@@ -37,6 +45,24 @@ target_link_libraries(pi_cuda
     cudadrv
 )
 
+if (MSVC)
+  # by defining __SYCL_BUILD_SYCL_DLL, we can use __declspec(dllexport)
+  # which are individually tagged for all pi* symbols in pi.h
+  target_compile_definitions(pi_cuda PRIVATE __SYCL_BUILD_SYCL_DLL)
+else()
+  # we set the visibility of all symbols 'hidden' by default.
+  # In pi.h file, we set exported symbols with visibility==default individually
+  target_compile_options(pi_cuda PUBLIC -fvisibility=hidden)
+
+  # This script file is used to allow exporting pi* symbols only.
+  # All other symbols are regarded as local (hidden)
+  set(linker_script "${CMAKE_CURRENT_SOURCE_DIR}/../ld-version-script.txt")
+
+  # Filter symbols based on the scope defined in the script file,
+  # and export pi* function symbols in the library.
+  target_link_libraries(pi_cuda PRIVATE "-Wl,--version-script=${linker_script}")
+endif()
+
```

Comment:
NIT: this code is duplicated in each plugin's CMakeLists.txt. Can't we move it to sycl/plugins/CMakeLists.txt

Reply:
Yes, it can be moved to sycl/plugins/CMakeLists.txt
I will make the change

Reply:
This change to the CMake was causing a build error. Probably due to operations like add_library, add_dependencies, set_target_properties, target_include_directories, and target_link_libraries not being executed yet. This operations are not the same across each plugin so can't be wrapped up in sycl/plugins/CMakeLists.txt easily. So, it may be more verbose than necessary but the keeping this code in the each plugin's CMakeList.txt is probably the simplest approach.

Reply:
I am OK with that.